arrows/ocv/moving_burnin_detector_opencv.cxx

```diff
@@ -0,0 +1,523 @@
+
+     if ( s > score )
+     {
+       center = ct;
+       score = s;
+     }
+    }
+  }
+
+  if ( score < cross_threshold )
+  {
+    center = cv::Point( 0, 0 );
+  }
+}
+
+// ----------------------------------------------------------------------------------
+void
+moving_burnin_detector_opencv::priv
+::draw_cross( cv::Mat& img, const cv::Point& center, cv::Scalar color, int width )
+{
+  // Middle cross
+  cv::line( img, center + cv::Point( cross_gap_x, 0 ), center +
+                          cv::Point( cross_gap_x + cross_length_x, 0), color, width );
+  cv::line( img, center + cv::Point( -cross_gap_x, 0 ), center +
+                          cv::Point( -cross_gap_x - cross_length_x, 0), color, width );
+  cv::line( img, center + cv::Point( 0, cross_gap_y ), center +
+                          cv::Point( 0, cross_gap_y + cross_length_y ), color, width );
+  cv::line( img, center + cv::Point( 0, -cross_gap_y ), center +
+                          cv::Point( 0, -cross_gap_y - cross_length_y ), color, width );
+
+  // Perpendicular ends
+  if( cross_ends_ratio > 0.0)
+  {
+    // Right
+    cv::line( img, center + cv::Point( cross_gap_x + cross_length_x,
+                                       static_cast<int>(-cross_ends_ratio * cross_length_x )),
+                   center + cv::Point( cross_gap_x + cross_length_x,
+                                       static_cast<int>( cross_ends_ratio * cross_length_x )), color, width );
+    // Left
+    cv::line( img, center + cv::Point(-cross_gap_x - cross_length_x+1,
+                                       static_cast<int>(-cross_ends_ratio * cross_length_x )),
+                   center + cv::Point(-cross_gap_x - cross_length_x+1,
+                                       static_cast<int>( cross_ends_ratio * cross_length_x )), color, width );
+    // Bottom
+    cv::line( img, center + cv::Point( static_cast<int>(-cross_ends_ratio * cross_length_y ),
+                                       cross_gap_y + cross_length_y ),
+                   center + cv::Point( static_cast<int>( cross_ends_ratio * cross_length_y ),
+                                       cross_gap_y + cross_length_y ), color, width );
+    // Top
+    cv::line( img, center + cv::Point( static_cast<int>(-cross_ends_ratio * cross_length_y ),
+                                      -cross_gap_y - cross_length_y ),
+                   center + cv::Point( static_cast<int>( cross_ends_ratio * cross_length_y ),
+                                      -cross_gap_y - cross_length_y ), color, width );
+  }
+}
+
+// ----------------------------------------------------------------------------------
+// main processing step
+kwiver::vital::image_container_sptr
+moving_burnin_detector_opencv::priv
+::filter( kwiver::vital::image_container_sptr input_image )
+{
+  // convert image to cv
+  cv::Mat cv_image = ocv::image_container::vital_to_ocv(
+    input_image->get_image(),
+    ocv::image_container::RGB_COLOR );
```

Comment:
I wonder if this `ocv::image_container::RGB_COLOR` and a bit later the `ocv::image_container::BGR_COLOR` might be giving you problems. First, they don't match, which is probably intentional but I figure I'd mention it anyway. Second, if you look at the implementation of these functions in `arrows/ocv/image_container.cxx`, `ocv_to_vital` doesn't use the `ColorMode` parameter **at all**, and `vital_to_ocv` only explicitly handles the `BGR_COLOR` case and lumps everything else into a bucket, not really doing anything with it. It looks to me whoever wrote these functions only really worked with BGR images, so I might investigate if they are actually doing what you hope they would be doing. Perhaps a cause of the pixel weirdness you were seeing?